### PR TITLE
[JAVA] Generate client files with multiple enum `oneOf` types

### PR DESCRIPTION
**Pull request by l3ender**
This change adds support for generating client files from multiple `oneOf` included items in an API definition. See https://github.com/OpenAPITools/openapi-generator/issues/12545 for full description/detail.

Fixes https://github.com/OpenAPITools/openapi-generator/issues/12545.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.0.1) (patch release), `6.1.x` (breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@bbdouglas @sreeshas @jfiala @lukoyanov @cbornet @jeff9finger @karismann @Zomzog @lwlee2608

Thank you!

### CODE REVIEW

**Comment by l3ender** [comment]
@wing328 could you please take a look at this PR and advise? I'm tagging you as I see you are active and have worked in the relevant area.

I have added test coverage for this fix and verified code generation is working as expected. Let me know if I can take any other actions. Thank you!

**Comment by wing328** [comment]
@l3ender thanks for the PR and sorry for the delay in reviewing as there're too many PRs.

I'll try to take a look tomorrow.

**Comment by wing328** [comment]
cc @OpenAPITools/generator-core-team

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 1]
```diff
@@ -2812,6 +2814,23 @@ protected void updateModelForComposedSchema(CodegenModel m, Schema schema, Map<S
         // end of code block for composed schema
     }
 
+    private void mergeProperties(Map<String, Schema> properties, Map<String, Schema> interfaceProperties) {
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java:2817]: can you please add a docstring explaining what this function does?

some comments in the code block below would be nice.

Comment (l3ender) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java]: Thank you for the review! I've updated and added some detail, which hopefully makes the addition clearer.

Thank you!

**Comment by wing328** [comment]
Overall it looks good. Will run more tests tomorrow and merge if no question from me or anyone. Thanks again for the PR.

**Comment by wing328** [comment]
I did some tests with the provided test spec and notice there's no change in java files but only changes (fixes) in openapi.yaml and doc as follows:
```diff
diff --git a/api/openapi.yaml b/api/openapi.yaml
index 9827f69..843ead4 100644
--- a/api/openapi.yaml
+++ b/api/openapi.yaml
@@ -38,6 +38,7 @@ components:
         type:
           enum:
           - dog
+          - cat
           type: string
       required:
       - type
diff --git a/docs/PetItems.md b/docs/PetItems.md
index 5c0e7ec..9b16294 100644
--- a/docs/PetItems.md
+++ b/docs/PetItems.md
@@ -15,6 +15,7 @@
 
 | Name | Value |
 |---- | -----|
+| DOG | &quot;dog&quot; |
 | CAT | &quot;cat&quot; |
```

PetItems.java already contains both Cat and Dog before this PR:
```java
            final TypeAdapter<Cat> adapterCat = gson.getDelegateAdapter(this, TypeToken.get(Cat.class));
            final TypeAdapter<Dog> adapterDog = gson.getDelegateAdapter(this, TypeToken.get(Dog.class));:
```

**Comment by l3ender** [comment]
Thanks for reviewing and helping merge, @wing328. I appreciate it!

**Pull request merged**